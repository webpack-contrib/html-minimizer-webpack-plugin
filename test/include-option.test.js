const HtmlMinimizerPlugin = require("../src/index");

const {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} = require("./helpers");

describe("include option", () => {
  let compiler;

  beforeEach(() => {
    const testHtmlId = "./include-exclude/*.html";

    compiler = getCompiler(testHtmlId);
  });

  it("should match snapshot for a single RegExp value include", async () => {
    new HtmlMinimizerPlugin({
      include: /include-exclude\/include/i,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.html$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple RegExp values include", async () => {
    new HtmlMinimizerPlugin({
      include: [
        /include-exclude\/include-(1|2)/i,
        /include-exclude\/include-(3|4)/i,
      ],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.html$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple String values include", async () => {
    new HtmlMinimizerPlugin({
      exclude: [
        "include-exclude/include-1",
        "include-exclude/include-2",
        "include-exclude/include-3",
        "include-exclude/include-4",
      ],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.html$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
