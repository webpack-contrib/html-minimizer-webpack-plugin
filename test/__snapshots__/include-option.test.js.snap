// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`include option should match snapshot for a single RegExp value include: assets 1`] = `
{
  "include-exclude/exclude-1.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/exclude-2.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/include-1.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-2.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-3.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-4.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
}
`;

exports[`include option should match snapshot for a single RegExp value include: errors 1`] = `[]`;

exports[`include option should match snapshot for a single RegExp value include: warnings 1`] = `[]`;

exports[`include option should match snapshot for multiple RegExp values include: assets 1`] = `
{
  "include-exclude/exclude-1.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/exclude-2.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/include-1.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-2.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-3.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-4.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
}
`;

exports[`include option should match snapshot for multiple RegExp values include: errors 1`] = `[]`;

exports[`include option should match snapshot for multiple RegExp values include: warnings 1`] = `[]`;

exports[`include option should match snapshot for multiple String values include: assets 1`] = `
{
  "include-exclude/exclude-1.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/exclude-2.html": "<!doctype html> <html lang="en"> <head> <meta charset="UTF-8"> <title>Document</title> </head> <body> <h1>My First Heading</h1> <p>My first paragraph.</p> <h2>An Unordered HTML List</h2> </body> </html> ",
  "include-exclude/include-1.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/include-2.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/include-3.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
  "include-exclude/include-4.html": "<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <title>Document</title>
</head>
<body>

<h1>My First Heading</h1>
<p>My first paragraph.</p>
<h2>An Unordered HTML List</h2>

</body>
</html>
",
}
`;

exports[`include option should match snapshot for multiple String values include: errors 1`] = `[]`;

exports[`include option should match snapshot for multiple String values include: warnings 1`] = `[]`;
